### PR TITLE
Mux reporter

### PR DESCRIPTION
**Pull request by kriskowal**
Initial support for reporting both JSON and List output.

### CODE REVIEW

**Comment by kriskowal** [comment]
Intent to rebase after #54, #56 land

File: output/list.go [hunk 1]
```diff
@@ -32,26 +33,33 @@ var green = color.New(color.FgGreen).SprintFunc()
...
+	return nil
+}
+
+func (l *List) Next(test execute.TestResponse, config *plan.Config) {
+	for _, result := range test.Results {
+		var statStr string
+		switch result.Status {
+		case execute.Success:
+			statStr = green("✓")
+			l.Summary.NumSuccess++
```
Comment (HelloGrayson) [inline output/list.go:50]: Would love to pull the book-keeping for summary out of the ReporterFunc - the way it is right now is that every reporter has to do this, which is lame.

File: output/report.go [hunk 2]
```diff
@@ -24,31 +24,37 @@ import (
...
 	// else a specific value was provided
 	switch name {
 	case "list":
-		return List, nil
+		return &List{}, nil
+	case "json":
+		return &JSON{}, nil
+	case "list,json":
+		fallthrough
+	case "json,list":
```
Comment (HelloGrayson) [inline output/report.go]: Yikes - hardcoding these?

Comment (kriskowal) [inline output/report.go]: Yeah, as an expedient. Maybe a temporary expedient, but need to give TTL attention before making this nice.

File: output/list.go (hunk 1 above)
Comment (kriskowal) [inline output/list.go:50]: #58

File: output/report.go [hunk 3]
```diff
@@ -29,29 +29,32 @@ import (
...
 	switch name {
 	case "list":
-		return List, nil
+		return &List{}, nil
 	case "json":
-		return JSON, nil
+		return &JSON{}, nil
+	case "list,json":
+		fallthrough
+	case "json,list":
```
Comment (prashantv) [inline output/report.go:51]: nit: instead of `fallthrough`, you can do

``` go
case "list,json", "json,list":
```

File: output/mux.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+type Mux struct {
+	s         Summary
+	Reporters []Reporter
+}
+
+func (r Mux) Start(config *plan.Config) error {
+	for _, reporter := range r.Reporters {
+		err := reporter.Start(config)
```
Comment (prashantv) [inline output/mux.go:15]: nit: inline into `if` below

File: main.go [hunk 5]
```diff
@@ -51,7 +51,20 @@ func main() {
 		fmt.Println(err)
 		os.Exit(1)
 	}
-	summary := reporter.Stream(config, results)
+
+	err = reporter.Start(config)
+	if err != nil {
```
Comment (prashantv) [inline main.go:56]: nit: use `err := reporter.Start` and inline into if. that way the `err` is scoped only to the body of the `if`

**Comment by kriskowal** [comment]
Consolidated in #59

File: output/report.go (hunk 3 above)
Comment (kriskowal) [inline output/report.go:51]: neat

**Pull request closed without merging**